### PR TITLE
[Enhance] Add `hparams` argument in `AutoAugment` and `RandAugment` and some other improvement.

### PR DESCRIPTION
**Pull request by mzr1996**
## Motivation

In `AutoAugment` and `RandAugment`, there are many common arguments in sub-policies. Now we have to set these arguments one by one.

## Modification

1. Add `hparams` argument in `AutoAugment` and `RandAugment`, so that we can specify common arguments by once.
2. Use `hparams` and [base variable](https://github.com/open-mmlab/mmcv/pull/1083) to simplify `RandAug` settings in swin-transformer.
2. Now `pad_val` supports using `Sequence` instead of only `tuple`.
3. Use smaller test images to speed up unit tests.

## BC-breaking (Optional)

No.

## Use cases (Optional)

```python
>>> policies = [
...     dict(
...         type='Translate',
...         magnitude_key='magnitude',
...         magnitude_range=(0, 1),
...         pad_val=128,
...         direction='horizontal'),
...     dict(
...         type='Rotate',
...         magnitude_key='angle',
...         magnitude_range=(0, 90))
... ]
>>> t = RandAugment(
...       policies=policies, 
...       num_policies=2, 
...       magnititude_level=12, 
...       hparams=dict(pad_val=15))
>>> # hparams won't override setting in policies config
>>> t.policies[0]['pad_val']
128
>>> # use hparams if not set in policies config
>>> t.policies[1]['pad_val']
15
```

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/datasets/pipelines/auto_augment.py [hunk 1]
```diff
@@ -9,12 +10,25 @@
 from ..builder import PIPELINES
 from .compose import Compose
 
+_HPARAMS_DEFAULT = dict(pad_val=128)
```
Comment (Ezra-Yu) [inline mmcls/datasets/pipelines/auto_augment.py:13]: Require comment.

File: configs/_base_/datasets/pipeline/rand_aug.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+rand_increasing_policies = [
```
Comment (Ezra-Yu) [inline configs/_base_/datasets/pipeline/rand_aug.py:1]: The  folder name should be ‘pipelines’.
Why using "increasing" in the name?

File: mmcls/datasets/pipelines/auto_augment.py [hunk 3]
```diff
@@ -9,12 +10,25 @@
...
+
 
 def random_negative(value, random_negative_prob):
     """Randomly negate value based on random_negative_prob."""
     return -value if np.random.rand() < random_negative_prob else value
 
 
+def merge_hparams(policy, hparams):
```
Comment (Ezra-Yu) [inline mmcls/datasets/pipelines/auto_augment.py:21]: Require function comment.

**Review by Ezra-Yu — COMMENTED** [review]
Do we need to provide auto_aug configuration file similar to rand_aug.py?

File: configs/_base_/datasets/pipeline/rand_aug.py (hunk 2 above)
Comment (mzr1996) [reply configs/_base_/datasets/pipeline/rand_aug.py]: `increasing` comes from `_RAND_INCREASING_TRANSFORMS` in timm. I have added a comment above that config.

**Comment by mzr1996** [comment]
> Do we need to provide auto_aug configuration file similar to rand_aug.py?

`AutoAugment` configuration will be added in ViT(#395)

**Review by Ezra-Yu — APPROVED** [review]
LGTM

**Pull request merged**